$MirOS: ports/net/curl/patches/patch-include_curl_mprintf_h,v 1.2 2008/08/26 19:16:10 tg Exp $
--- include/curl/mprintf.h.orig	Tue May 20 12:21:53 2008
+++ include/curl/mprintf.h	Tue Aug 26 19:04:54 2008
@@ -32,18 +32,46 @@
 extern "C" {
 #endif
 
-CURL_EXTERN int curl_mprintf(const char *format, ...);
-CURL_EXTERN int curl_mfprintf(FILE *fd, const char *format, ...);
-CURL_EXTERN int curl_msprintf(char *buffer, const char *format, ...);
+CURL_EXTERN int curl_mprintf(const char *format, ...)
+    __attribute__((__nonnull__(1)))
+    __attribute__((__format__(__printf__, 1, 2)));
+CURL_EXTERN int curl_mfprintf(FILE *fd, const char *format, ...)
+    __attribute__((__nonnull__(1)))
+    __attribute__((__nonnull__(2)))
+    __attribute__((__format__(__printf__, 2, 3)));
+CURL_EXTERN int curl_msprintf(char *buffer, const char *format, ...)
+    __attribute__((__nonnull__(1)))
+    __attribute__((__nonnull__(2)))
+    __attribute__((__format__(__printf__, 2, 3)));
 CURL_EXTERN int curl_msnprintf(char *buffer, size_t maxlength,
-                               const char *format, ...);
-CURL_EXTERN int curl_mvprintf(const char *format, va_list args);
-CURL_EXTERN int curl_mvfprintf(FILE *fd, const char *format, va_list args);
-CURL_EXTERN int curl_mvsprintf(char *buffer, const char *format, va_list args);
+                               const char *format, ...)
+    __attribute__((__format__(__printf__, 3, 4)))
+    __attribute__((__nonnull__(1)))
+    __attribute__((__nonnull__(3)))
+    __attribute__((__bounded__(__string__, 1, 2)));
+CURL_EXTERN int curl_mvprintf(const char *format, va_list args)
+    __attribute__((__nonnull__(1)))
+    __attribute__((__format__(__printf__, 1, 0)));
+CURL_EXTERN int curl_mvfprintf(FILE *fd, const char *format, va_list args)
+    __attribute__((__nonnull__(1)))
+    __attribute__((__nonnull__(2)))
+    __attribute__((__format__(__printf__, 2, 0)));
+CURL_EXTERN int curl_mvsprintf(char *buffer, const char *format, va_list args)
+    __attribute__((__nonnull__(1)))
+    __attribute__((__nonnull__(2)))
+    __attribute__((__format__(__printf__, 2, 0)));
 CURL_EXTERN int curl_mvsnprintf(char *buffer, size_t maxlength,
-                                const char *format, va_list args);
-CURL_EXTERN char *curl_maprintf(const char *format, ...);
-CURL_EXTERN char *curl_mvaprintf(const char *format, va_list args);
+                                const char *format, va_list args)
+    __attribute__((__nonnull__(1)))
+    __attribute__((__nonnull__(3)))
+    __attribute__((__format__(__printf__, 3, 0)))
+    __attribute__((__bounded__(__string__, 1, 2)));
+CURL_EXTERN char *curl_maprintf(const char *format, ...)
+    __attribute__((__nonnull__(1)))
+    __attribute__((__format__(__printf__, 1, 2)));
+CURL_EXTERN char *curl_mvaprintf(const char *format, va_list args)
+    __attribute__((__nonnull__(1)))
+    __attribute__((__format__(__printf__, 1, 0)));
 
 #ifdef _MPRINTF_REPLACE
 # undef printf
